src/model/Builtin.js

```diff
@@ -0,0 +1,132 @@
+import assert from 'assert'
+import {Ref} from '../objects'
+import * as query from '../query'
+import {RefCodec} from './Codec'
+import Model from './Model'
+
+/**
+ * Builtins are special classes that exist by default.
+ * If you want to store custom data, you can add new fields with e.g. `Database.addField`.
+ */
+export default class Builtin extends Model {
+  /** @private */
+  static setup(...args) {
+    super.setup(...args)
+    assert(!this.isAbstract())
+    // classRef does not have 'classes' in front
+    this.classRef = new Ref(this.faunaClassName)
+    for (const fieldName in this.fields)
+      // Builtin fields do not have 'data' in front of path
+      this.fields[fieldName].path = [fieldName]
+  }
+}
+
+/** See the [docs](https://faunadb.com/documentation/objects#databases). */
+export class Database extends Builtin { }
+Database.setup('databases', {name: {}, api_version: {}})
+
+/** See the [docs](https://faunadb.com/documentation/objects#keys). */
+export class Key extends Builtin { }
+Key.setup('keys', {
+  database: {codec: RefCodec},
+  role: {},
+  secret: {},
+  hashed_secret: {}
+})
+
+/** See the [docs](https://faunadb.com/documentation/objects#classes). */
+export class Class extends Builtin {
+  /** Creates a class for the {@link Model}. */
+  static async createForModel(client, modelClass, data={}) {
+    return await this.create(client, Object.assign({name: modelClass.faunaClassName}, data))
+  }
+
+  /** Gets the class associated with a {@link Model}. */
+  static async getForModel(client, modelClass) {
+    try {
+      await this.get(client, modelClass.classRef)
+    } catch (err) {
+      console.log(err)
```

Comment:
Oops, debug code got left in. Usually I remember to leave a comment so the linter will remind me to delete these.